src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -0,0 +1,380 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp::Ordering;
+use std::future::Future;
+
+use risingwave_hummock_sdk::key_range::KeyRange;
+use risingwave_hummock_sdk::VersionedComparator;
+use risingwave_pb::hummock::SstableInfo;
+
+use crate::hummock::compactor::sstable_store::SstableBlocks;
+use crate::hummock::compactor::CompactorSstableStoreRef;
+use crate::hummock::iterator::{Forward, HummockIterator};
+use crate::hummock::sstable_store::TableHolder;
+use crate::hummock::value::HummockValue;
+use crate::hummock::{BlockHolder, BlockIterator, HummockResult};
+use crate::monitor::StoreLocalStatistic;
+
+pub struct SstablePrefetchIterator {
+    /// The iterator of the current block.
+    block_iter: Option<BlockIterator>,
+
+    /// Current block index.
+    cur_idx: usize,
+
+    /// Reference to the sst
+    sst: TableHolder,
+
+    blocks: SstableBlocks,
```

Comment:
This can be replaced with the read stream after we have streaming read supported in object store. cc @ALeitert

Reply:
Yes.